src/main/scala/com/softwaremill/mqperf/mq/ArtemisMq.scala

```diff
@@ -0,0 +1,18 @@
+package com.softwaremill.mqperf.mq
+
+import javax.jms.ConnectionFactory
+
+import com.softwaremill.mqperf.config.TestConfig
+import org.apache.activemq.artemis.jms.client.ActiveMQConnectionFactory
+
+class ArtemisMq(val testConfig: TestConfig) extends Jms2Mq {
+
+  override lazy val connectionFactory: ConnectionFactory = {
+    val hosts = "failover:(" + testConfig.brokerHosts.map(h => s"tcp://$h:61616").mkString(",") + ")"
+    val cf = new ActiveMQConnectionFactory(hosts)
+    cf.setConfirmationWindowSize(10485760)
+    cf.setBlockOnDurableSend(false)
```

Comment:
No because we are using async sending and relying on the completion listener. As you note in the completion listener code we are checking all the messages sent are sent, this logic can become more detailed, but for this test this is enough.

This is the same as kafka, where by on send it isn't actually sent to the broker, it is only sent sometime in the future within a batch, as such you rely in kafka also on completion listener or the returned future of the send.

This is making this the same (apples to apples)

The reason this is defaulted is because many people will code in a non async fashion, because of legacy JMS 1.1